find-mmc-bypartlabel

```diff
@@ -20,28 +20,28 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
 # Pass the partition label name as $1
-if test -z $1; then
+if [ -z "$1" ]; then
 	echo "$0: Error, no label name given!" > /dev/kmsg
 	exit 1
 fi
 
-if ! test -d /sys/class/block; then
+if [ ! -d /sys/class/block ]; then
 	echo "$0: Error, please mount sysfs" > /dev/kmsg
 	exit 1
 fi
 
-while [ ! -d /sys/class/block/mmcblk0p* ]; do
+while [ ! -d /sys/class/block/mmcblk0p* ] && [ ! -d /sys/class/block/sda ] ; do
```

Comment:
when I use 
```
while [ ! -d /sys/class/block/mmcblk0p* ] && [ ! -d /sys/class/block/sd* ] ; do
```

I get "line 33: [: too many arguments" when running in booted shell. I suggest that we should make a clean script without any errors like it. possibly original one was doing the same, though.